### PR TITLE
Fix `cbsceneinventory` not working correctly with "filter outdated" enabled

### PR DESCRIPTION
**Pull request by BigRoy**
This fixes the "update to latest" not working when the 'outdated filter' was enabled.

### CODE REVIEW

File: avalon/tools/cbsceneinventory/proxy.py [hunk 1]
```diff
@@ -63,6 +63,13 @@ def filterAcceptsRow(self, source_row, source_parent):
         column = self.filterKeyColumn()
         index = model.index(source_row, column, source_parent)
 
+        # Filters below are per version group (top-level entries) so if
```
Comment (mottosso) [inline avalon/tools/cbsceneinventory/proxy.py]: Hm, I can't make sense out what this comment means. Do you think you can reformulate it?

Comment (BigRoy) [inline avalon/tools/cbsceneinventory/proxy.py]: https://github.com/getavalon/core/pull/226/commits/b62bc134aaefba0629c0f71ff912279ba6422a8e tries to reformulate it.

**Comment by mottosso** [comment]
That makes more sense to me. What do you think?

**Comment by BigRoy** [comment]
We could also do this:
```python
        ...

        # Filter the "representation" group by whether they are outdated
        is_representation_group = not index.parent().isValid()
        if is_representation_group:

            # Filter to those that have the different version numbers
            node = index.internalPointer()
            version = node.get("version", None)
            highest = node.get("highest_version", None)

            # Always allow indices that have no version data at all
            if version is None and highest is None:
                return True

            # If either a version or highest is present but not the other
            # consider the item invalid
            if version is None or highest is None:
                return False

            return version != highest

        # If an index hasn't been filtered, consider it valid
        return True
```

But I think it's good as is.

**Pull request merged**